src/python/gudhi/representations/vector_methods.py

```diff
@@ -350,6 +351,92 @@ def __call__(self, diag):
         """
         return self.fit_transform([diag])[0,:]
 
+
+class IrregularBettiCurve(BaseEstimator, TransformerMixin):
```

Comment:
I would be in favor of having just one `BettiCurve` class, with a potential parameter `even` or `regular` that would specify whether we sample uniformly or not.

Reply:
That's a good idea.